### DISCUSSION TITLE
Over write the parent table primark key by relation table whe use has on relation

### DISCUSSION THREAD

**Discussion by hasib2008** [yajra/laravel-datatables#2888] [General]
Overwrite the parent table primary key (id) by relation table when use has on the relation 
I have 2 table 

**1) company_user** 
 id -> primary key 
company_id  ->foreign key of company table
users_id    -> foreign key of the user table
status 

**users** (Laravel Default )
so when I  tun 

$models = CompanyUsers::myCompany()->has('users')->with('users');
return Datatables::eloquent($models)
    ->addIndexColumn()
     ->addColumn('check_box', function ($model) {
         return view('common.checkbox', ['model' => $model, 'class' => 'i-checks people-checkbox']);
     })
    ->addColumn('name', function ($model) {
        return $model->users->name;
   })
    ->addColumn('status', 'common.status')
    ->addColumn('action', templatePath('users.action'))
    ->rawColumns(['action', 'status','name' ,'check_box'])
    ->make(true);
 
 if I try to access $mode->id it gives me the user table id not the company user table 
 
 
<img width="306" alt="image" src="https://user-images.githubusercontent.com/8968706/197262386-5a884b2e-56d2-4828-9540-d221e8ac2b45.png">

**Comment by hasib2008** [comment]
@yajra Can you please suggest

**Comment by Arne1303** [comment]
I don't know what's going on in `myCompany()` but as far as I can read the query you should get the ID of the company user by calling `->id`

If you need the ID of an other model you can always call the relationship to get it

**Comment by hasib2008** [comment]
@Arne1303 **myCompany()** is not just adding a where condition I also try by removing this but the same issue 
No, I get the primary key of the parent model in this case **company_users** table  not relation model

in my **company_users** table only one record right now and the primary key is 4
<img width="176" alt="image" src="https://user-images.githubusercontent.com/8968706/197585820-0f4071e0-2845-49b3-869e-8762d4c1661b.png">

And here is the response where the primary is overwritten by the user table primary key also notice it does not work $hidden as you can see   'password', and 'remember_token' also display  

if I try to get its id by **->id** it gives me 1 but it should be 4

<img width="505" alt="image" src="https://user-images.githubusercontent.com/8968706/197586400-30e150fa-fd25-40da-b81a-dcdeb1fa1ad5.png">

I use sort by **users.first_name**

<img width="958" alt="image" src="https://user-images.githubusercontent.com/8968706/197588898-802d86fd-cd98-4117-b36c-3da198ccd300.png">

**Comment by Arne1303** [comment]
Can you send the user relation definition in user company?

**Comment by hasib2008** [comment]
@Arne1303 
here is the relation details.

<img width="362" alt="image" src="https://user-images.githubusercontent.com/8968706/197590499-a1e8de38-7e03-4cbf-9a02-f7f6dbb869b2.png">

**Comment by Arne1303** [comment]
I can't find anything wrong with it, it should give you the ID of the company user. You could try specifying the table in a select statement and check if you are really viewing the right database in your DBMS

**Comment by yajra** [comment]
I think you just need to specify a select SQL:

```php
CompanyUsers::myCompany()->select('company_users.*')->has('users')->with('users');
```

**Comment by hasib2008** [comment]
> I think you just need to specify a select SQL:

Yes after doing that it works
